xds/internal/balancer/clusterimpl/clusterimpl.go

```diff
@@ -105,13 +103,17 @@ type clusterImplBalancer struct {
 	clusterNameMu sync.Mutex
 	clusterName   string
 
-	serializer       *grpcsync.CallbackSerializer
-	serializerCancel context.CancelFunc
-
-	// childState/drops/requestCounter keeps the state used by the most recently
-	// generated picker.
+	childStateMu sync.Mutex // guards childState and inhibitPickerUpdates
+	// Set during UpdateClientConnState when pushing updates to child policy.
+	// Prevents state updates from child policy causing new pickers to be sent
+	// up the channel. Cleared after child policy have processed the
+	// updates sent to them, after which a new picker is sent up the channel.
+	inhibitPickerUpdates bool
```

Comment:
Github doesn't allow you to comment on unchanged lines of code. So, this comment actually is for the line that defines the `clusterImplBalancer` struct. 

We have a bunch of fields in this struct now without clear documentation about when/where they are set, and when/where they are read (or updated), and what lock (if at all, any) should be held when accessing them.

I'd like to see something like this:
```golang
type clusterImplBalancer struct {
	balancer.ClientConn

	// The following fields are set at creation time, and are read-only after
	// that, and therefore need not be protected by a mutex.
	logger      *grpclog.PrefixLogger
	loadWrapper *loadstore.Wrapper

	// The following fields are only accessed from balancer API methods, which
	// are guaranteed to be called serially by gRPC.
	xdsClient        xdsclient.XDSClient     // Sent down in ResolverState attributes.
	cancelLoadReport func()                  // To stop reporting load through the above xDS client.
	edsServiceName   string                  // EDS service name to report load for.
	lrsServer        *bootstrap.ServerConfig // Load reporting server configuration.
	dropCategories   []DropConfig            // The categories for drops.
	child            *gracefulswitch.Balancer

	// The following fields are protected by mu, since they are accessed in
	// balancer API methods and in methods called from the child policy.
	mu                    sync.Mutex
	clusterName           string                            // The cluster name for credentials handshaking.
	inhibitPickerUpdates  bool                              // Inhibits state updates from child policy when processing an update from the parent.
	childState            balancer.State                    // Most recent state update from the child policy.
	drops                 []*dropper                        // Drops implementation.
	requestCounterCluster string                            // The cluster name for the request counter, from LB config.
	requestCounterService string                            // The service name for the request counter, from LB config.
	requestCountMax       uint32                            // Max concurrent requests, from LB config.
	requestCounter        *xdsclient.ClusterRequestsCounter // Tracks total inflight requests for a given service.
	telemetryLabels       map[string]string                 // Telemetry labels to set on picks, from LB config.
}
```

Also, note that I got rid of the `bOpts` and `config` fields, because they are not required any more.

Reply:
Sure, Updated.